test/parallel_api/algorithm/alg.sorting/alg.binary.search/binary.search/binary_search.pass.cpp

```diff
@@ -104,8 +104,10 @@ DEFINE_TEST(test_binary_search)
     {
         typedef typename ::std::iterator_traits<Iterator1>::value_type ValueT;
 
+        TestDataTransfer<UDTKind::eKeys, Size> host_keys(*this, n);
+
         // call algorithm with no optional arguments
-        initialize_data(first, value_first, result_first, n);
+        initialize_data(get_fill_keys(host_keys, first, n), value_first, result_first, n);
```

Comment:
```is_equal_exist''' is funtion. They are implemented in the file https://github.com/oneapi-src/oneDPL/pull/850/files#diff-70c56b162d899deb7e7f71142293bafd9f927fa0bec27c05317cb5955c5a956d.
The main goal is to return non-const iterator for fill source key data. Because now source key data may be described in the tests by const iterator.

Reply:
The function ```get_fill_keys``` renamed into ```get_non_const_it```